test/metrics/kafka-partition-metrics.td

```diff
@@ -0,0 +1,46 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+#
+# Test per partition kafka metrics on initial and dynamically added partitions
+#
+
+$ kafka-create-topic topic=input_csv partitions=3
+
+> CREATE MATERIALIZED SOURCE input_csv (first, second)
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-input_csv-${testdrive.seed}' WITH (topic_metadata_refresh_interval_ms=5000)
+  FORMAT CSV WITH 2 COLUMNS;
+
+$ kafka-ingest format=bytes topic=input_csv partition=0
+1,2
+2,3
+
+$ kafka-ingest format=bytes topic=input_csv partition=1
+3,4
+4,5
+
+$ kafka-ingest format=bytes topic=input_csv partition=2
+5,6
+
+
+> SELECT * from input_csv;
+first second mz_offset
+----------------------
+1     2      1
+2     3      2
+3     4      1
+4     5      2
+5     6      1
+
+> SELECT labels->>'partition_id' AS partition_id, max(value) as value FROM mz_metrics WHERE metric='mz_kafka_partition_offset_max' group by partition_id;
+partition_id value
+------------------
+0              2
+1              2
+2              1
```

Comment:
This test file is nearly a strict subset of the other test script. Doesn't seem worth having both to me? (Could add multiple initial partitions to the other file, and then it truly would be a strict superset of this script.)

Reply:
The observed behavior until I fixed the bugs was different when it started with 3 partitions vs growing from 1 to 3. As a result my thoughts were that having a test of fixed partition count and variable count is worthwhile but I can remove it if you think its redundant.